src/textual/reactive.py

```diff
@@ -67,204 +159,159 @@ def __init__(
         self._repaint = repaint
         self._init = init
         self._always_update = always_update
-        self._run_compute = compute
+        self._trigger_object_compute = compute
 
     def __rich_repr__(self) -> rich.repr.Result:
         yield self._default
         yield "layout", self._layout
         yield "repaint", self._repaint
         yield "init", self._init
         yield "always_update", self._always_update
-        yield "compute", self._run_compute
+        yield "compute", self._trigger_object_compute
 
-    @classmethod
-    def init(
-        cls,
-        default: ReactiveType | Callable[[], ReactiveType],
-        *,
-        layout: bool = False,
-        repaint: bool = True,
-        always_update: bool = False,
-        compute: bool = True,
-    ) -> Reactive:
-        """A reactive variable that calls watchers and compute on initialize (post mount).
+    def __set_name__(self, owner: Type[Reactable], name: str) -> None:
+        # The name of the attribute
+        self.name = name
+        # The internal name where the attribute's value is stored
+        self.internal_name = f"_reactive_{name}"
+        # Corresponding compute, watch and validate methods, if exist
+        # NOTE that these are all instance methods, but are stored here
+        # bound to the reactable class (and not the instance, which at
+        # this point is unknown.). This means that invoking them must
+        # supply a `self` parameter as well.
+        self.compute_function = getattr(owner, f"compute_{name}", None)
```

Comment:
caching the actual methods (bound to the class at this point, but read the comment above please!) so save all the duplicate `f"_foo_{name}"` and `getattr()` logic.